packages/main/src/methods/chat-session-helpers.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Content, POSSIBLE_ROLES, Part, Role } from "../../types";
+import { GoogleGenerativeAIError } from "../errors";
+
+const VALID_PART_FIELDS: Array<keyof Part> = [
+  "text",
+  "inlineData",
+  "functionCall",
+  "functionResponse",
+];
+
+const VALID_PARTS_PER_ROLE: { [key in Role]: Array<keyof Part> } = {
+  user: ["text", "inlineData"],
+  function: ["functionResponse"],
+  model: ["text", "functionCall"],
+};
+
+const VALID_PREVIOUS_CONTENT_ROLES: { [key in Role]: Role[] } = {
+  user: ["model"],
+  function: ["model"],
+  model: ["user", "function"],
+};
+
+export function validateChatHistory(history: Content[]): void {
+  let prevContent: Content;
+  for (const currContent of history) {
+    const { role, parts } = currContent;
+    if (!prevContent && role !== "user") {
+      throw new GoogleGenerativeAIError(
+        "First content should be with role 'user'",
+      );
+    }
+    if (!POSSIBLE_ROLES.includes(role)) {
+      throw new GoogleGenerativeAIError("Each item should include role field");
```

Comment:
Can we list the valid roles here? Like
```
`Each item should include a role field. Valid roles: ${POSSIBLE_ROLES.join(',')}`
```
or something. In case they did include a role but they mistyped it or it doesn't exist (like "system" at the moment).